core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3511,4 +3501,14 @@ object KafkaApis {
     FetchResponse.sizeOf(versionId, unconvertedResponse.responseData.entrySet
       .iterator.asScala.filter(element => quota.isThrottled(element.getKey)).asJava)
   }
+
+  // visible for testing
+  private[server] def shouldNeverReceive(request: RequestChannel.Request): Exception = {
```

Comment:
Why move these and not `unsupported`/`notYetSupported`?

Reply:
Ah, I hadn't tested for those, and I moved based on rewriting tests.  Will move.